include/deal.II/lac/relaxation_block.templates.h

```diff
@@ -94,66 +94,78 @@ inline
 void
 RelaxationBlock<MatrixType, InverseNumberType, VectorType>::invert_diagblocks ()
 {
-  const MatrixType &M=*A;
-  FullMatrix<InverseNumberType> M_cell;
-
   if (this->same_diagonal())
     {
       Assert(false, ExcNotImplemented());
     }
   else
     {
-      for (size_type block=0; block<additional_data->block_list.n_rows(); ++block)
-        {
-          const size_type bs = additional_data->block_list.row_length(block);
-          M_cell.reinit(bs, bs);
+      // compute blocks in parallel
+      parallel::apply_to_subranges(0, this->additional_data->block_list.n_rows(),
+                                   std_cxx11::bind(&RelaxationBlock<MatrixType, InverseNumberType, VectorType>::block_kernel, this,
+                                                   std_cxx11::_1, std_cxx11::_2),
+                                   16);
+    }
+  this->inverses_computed(true);
+}
 
-          // Copy rows for this block
-          // into the matrix for the
-          // diagonal block
-          SparsityPattern::iterator row
-            = additional_data->block_list.begin(block);
-          for (size_type row_cell=0; row_cell<bs; ++row_cell, ++row)
-            {
-//TODO:[GK] Optimize here
```

Comment:
I actually asked @guidokanschat if this should be left there and he said no. Should I still put it back?